internal/brokers/examplebroker.go

```diff
@@ -77,17 +77,35 @@ const (
 func newExampleBroker(name string) (b *exampleBroker, fullName, brandIcon string) {
 	return &exampleBroker{
 		currentSessions:        make(map[string]sessionInfo),
-		currentSessionsMu:      sync.Mutex{},
+		currentSessionsMu:      sync.RWMutex{},
 		userLastSelectedMode:   make(map[string]string),
 		userLastSelectedModeMu: sync.Mutex{},
 		isAuthorizedCalls:      make(map[string]isAuthorizedCtx),
 		isAuthorizedCallsMu:    sync.Mutex{},
 	}, strings.ReplaceAll(name, "_", " "), fmt.Sprintf("/usr/share/brokers/%s.png", name)
 }
 
-// GetAuthenticationModes returns the list of supported authentication modes for the selected broker depending on user name.
-func (b *exampleBroker) GetAuthenticationModes(ctx context.Context, username, lang string, supportedUILayouts []map[string]string) (sessionID, encryptionKey string, authenticationModes []map[string]string, err error) {
+// NewSession creates a new session for the specified user.
+func (b *exampleBroker) NewSession(ctx context.Context, username, lang string) (sessionID, encryptionKey string, err error) {
 	sessionID = uuid.New().String()
```

Comment:
Should we loop here until we guarantee that the ID is actually unique?

Reply:
I wouldn’t, it’s guarranteed to be a uuid by the library.